be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -973,11 +972,20 @@ Status SegmentIterator::_init_inverted_index_iterators() {
     }
     for (auto cid : _schema.column_ids()) {
         int32_t unique_id = _schema.unique_id(cid);
+        auto column_name = _schema.column(cid)->name();
         if (_inverted_index_iterators.count(unique_id) < 1) {
             RETURN_IF_ERROR(_segment->new_inverted_index_iterator(
                     _opts.tablet_schema->column(cid), _opts.tablet_schema->get_inverted_index(cid),
                     _opts.stats, &_inverted_index_iterators[unique_id]));
         }
+
+        if (_inverted_index_iterators.count(unique_id) > 0
```

Comment:
How about if an user does not add index, but wants to use match?

Reply:
will use parser type: `InvertedIndexParserType::PARSER_UNKNOWN`, which default use `SimpleAnalyzer`